sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/om/Builder.java

```diff
@@ -375,6 +378,23 @@ public boolean build() {
 						getLogger().log("Exception occured in file " + sourceFile.getAbsolutePath() + ", preprocess failed.", 3, t);
 						handleBuildException(t);
 					} finally {
+                                                if (outputFile.exists()) {
+							try {
+								// Load FileAttribute.Tag class and grab constructor
+								Class<?> fileAttributeClass = Class.forName("com.ibm.jzos.FileAttribute");
+								Class<?> fileAttributeTagClass = Class.forName("com.ibm.jzos.FileAttribute$Tag");
+								Constructor<?> constructor = fileAttributeTagClass.getConstructor(char.class, boolean.class);
+
+								Field ccsid = fileAttributeTagClass.getField("CCSID_IBM_1047");
+								char ccsidChar = (char) ccsid.get(null);
+								Object tag = constructor.newInstance(ccsidChar, true);
+
+								Method setTag = fileAttributeClass.getMethod("setTag", String.class, fileAttributeTagClass);
```

Comment:
The class and reflection lookups should be done once only, not once for each file.